src/guild/ShoukakuPlayer.js

```diff
@@ -437,7 +437,6 @@ class ShoukakuPlayer extends EventEmitter {
      * @private
      */
     _onPlayerEvent(json) {
-        this.position = 0;
```

Comment:
this.position must reset, but seeing it affects reconnects, check what event fires upon reconnect so we can exempt that on putting this.position to zero

Reply:
> ShoukakuSocket.js 171:49
![image](https://user-images.githubusercontent.com/67855694/131214001-15dde4f3-9fe1-4515-9ce6-b4f662c3e15e.png)

> ShoukakuSocket.js 292:41
![image](https://user-images.githubusercontent.com/67855694/131213860-f4a1a978-4c8f-4d02-bcf0-decfe510cfea.png)

> ShoukakuPlayer.js 428:18
![image](https://user-images.githubusercontent.com/67855694/131213836-5035589b-810a-40da-8c0e-5ae3c3ae91e4.png)

Reply:
For starters, if json.type in shoukaku player onPlayerEvent is not equal to track start event, reset this.position to zero, see if this fixes your issue on 
> I think i get some bug, when lavalink server dies, track position returns to 0. It's normal?

Reply:
why we need to reset this.position to zero on PlayerEvent?

Reply:
If lavalink dies, lavalink will sendTrackExceptionEvent before die
![image](https://user-images.githubusercontent.com/67855694/131215047-1078d7a6-2a68-4728-98c0-e4582ca92061.png)
![image](https://user-images.githubusercontent.com/67855694/131214999-5f630d0d-f9b2-4598-99fe-60e147252916.png)

Reply:
> For starters, if json.type in shoukaku player onPlayerEvent is not equal to track start event, reset this.position to zero, see if this fixes your issue on
>> I think i get some bug, when lavalink server dies, track position returns to 0. It's normal?

Ahh, sorry for my misunderstanding before, I think just type PlayerExceptionEvent and WebSocketClosedEvent

Reply:
put the position on zero in trackstart